ddtrace/tracer/sqlcomment.go

```diff
@@ -188,7 +190,53 @@ func commentQuery(query string, tags map[string]string) string {
 	return b.String()
 }
 
-// Extract is not implemented on SQLCommentCarrier
+// Extract parses for key value attributes in a sql query's comment in order to build a span context
 func (c *SQLCommentCarrier) Extract() (ddtrace.SpanContext, error) {
-	return nil, nil
+	var ctx spanContext
+	re := regexp.MustCompile(`/\*(.*?)\*/`) // extract sql comment
+	if match := re.FindStringSubmatch(c.Query); len(match) == 2 {
+		comment := match[1]
+		kvs := strings.Split(comment, ",")
+		for _, unparsedKV := range kvs {
+			if splitKV := strings.Split(unparsedKV, "="); len(splitKV) == 2 {
+				key := splitKV[0]
+				value := strings.Trim(splitKV[1], "'")
+				switch key {
+				case sqlCommentTraceParent:
+					traceID, spanID, sampled, err := decodeTraceParent(value)
+					if err != nil {
+						return nil, err
+					}
+					ctx.traceID.SetLower(traceID)
+					ctx.spanID = spanID
+					ctx.setSamplingPriority(sampled, samplernames.Unknown)
+				default:
+				}
+			} else {
+				return nil, ErrSpanContextCorrupted
+			}
+		}
+	} else {
+		return nil, ErrSpanContextNotFound
+	}
+	if ctx.traceID.Empty() || ctx.spanID == 0 {
+		return nil, ErrSpanContextNotFound
+	}
+	return &ctx, nil
+}
+
+// decodeTraceParent decodes trace parent as per the w3c trace context spec (https://www.w3.org/TR/trace-context/#version).
+func decodeTraceParent(traceParent string) (traceID uint64, spanID uint64, sampled int, err error) {
+	if splitParent := strings.Split(traceParent, "-"); len(splitParent) == 4 {
```

Comment:
Because the trace parent standard has a fixed number of bytes for each component, it might be faster and make more sense to skip the split and just extract the range of chars for each component.